src/core/mesh/qgsmeshdataprovidertemporalcapabilities.cpp

```diff
@@ -132,7 +132,8 @@ QgsDateTimeRange QgsMeshDataProviderTemporalCapabilities::timeExtent( const QDat
     if ( !times.isEmpty() )
     {
       durationSinceFirst += times.first();
-      durationSinceLast += times.last();
+      // adding one timestep (length as: times.last() - times.beforeLast)), to be sure we have enough steps
+      durationSinceLast += times.last() + ( times.last() - times[times.length() - 2] );
```

Comment:
Note that having data with just 1 timestep, the 'times' here will be empty, so you will only enter when length is at least 2 so times[times.length() - 2] is safe.